File: Array&Hashing/twoSum/problem.js

 function twoSum(nums, target) {

 // __________________(Brute force)__________________________

//   for(let i=0; i<=nums.length -1; i++ ){
//     for(let j=0; j<=nums.length-1; j++){
//         if(nums[i] + nums[j] === target && i !== j){
//             return [i,j];
//         }

//     }
//   }

 // _______________________(Hash maping)______________________


  let hash = {};

  for(let i=0; i<=nums.length-1; i++){
      let diff = target - nums[i];
      if(hash[diff] !== undefined){
          return [hash[diff],i];
        }  else{
        }
        hash[nums[i]] = i;     
        console.log(hash);
         
  }

return 0;

 }

 console.log(twoSum([3,4,5,6],8));
 

File: Array&Hashing/twoSum/problem.py
#_____________________Hash Maping__________________________________*


# def twoSum(nums, target):
#      h={}
     

#      for j ,k in enumerate(nums):
   
#             check= target - k
#             if(check in h and h[check]!=j):
#                  return h[check],j
#             else:
#                   h[k]=j
                 
#      return []
                
                   

                

# nums=[3,4,5,6]
# target=7

# print(twoSum(nums,target))

#_________________( Brute Force )_______________________*


def twoSum(nums, target):
    lenght=len(nums)
    for i in range(lenght):
        for j in range(lenght):
            if(nums[i]+nums[j]==target and i!=j):
                return [i,j]
            
    return []
    

nums=[3,4,5,6]
target=7

print(twoSum(nums,target))

